components/gcp/dataproc/submit_hadoop_job/component.yaml

```diff
@@ -0,0 +1,81 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: dataproc_submit_hadoop_job
+description: >-
+  Submits a Cloud Dataproc job for running Apache Hadoop MapReduce jobs on 
+  Apache Hadoop YARN.
+inputs:
+  - name: project_id
+    description: >-
+      Required. The ID of the Google Cloud Platform project that the cluster
+      belongs to.
+  - name: region
+    description: >-
+      Required. The Cloud Dataproc region in which to handle the request.
+  - name: cluster_name
+    description: 'Required. The cluster to run the job.'
+  - name: main_jar_file_uri
+    description: >-
+      The HCFS URI of the jar file containing the main class. Examples: 
+      `gs://foo-bucket/analytics-binaries/extract-useful-metrics-mr.jar` 
+      `hdfs:/tmp/test-samples/custom-wordcount.jar` 
+      `file:///home/usr/lib/hadoop-mapreduce/hadoop-mapreduce-examples.jar`
+  - name: main_class
+    description: >-
+      The name of the driver's main class. The jar file 
+      containing the class must be in the default CLASSPATH or specified 
+      in `jarFileUris`.
+  - name: args
+    default: ''
+    description: >-
+      Optional. The arguments to pass to the driver. Do not include 
+      arguments, such as -libjars or -Dfoo=bar, that can be set as job properties, 
+      since a collision may occur that causes an incorrect job submission.
+  - name: hadoop_job
+    default: ''
+    description: >-
+      Optional. The full payload of a
+      [hadoop job](https://cloud.google.com/dataproc/docs/reference/rest/v1/HadoopJob).
+  - name: job
+    default: ''
+    description: >-
+      Optional. The full payload of a
+      [Dataproc job](https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs).
+  - name: wait_interval
+    default: '30'
+    description: >-
+      Optional. The wait seconds between polling the operation.
+      Defaults to 30.
+outputs:
+  - {name: job_id, description: 'The ID of the created job.'}
+implementation:
+  container:
+    image: gcr.io/ml-pipeline/ml-pipeline-gcp:latest
+    args: [
+      kfp_component.google.dataproc, submit_hadoop_job,
+      --project_id, {inputValue: project_id},
+      --region, {inputValue: region},
+      --cluster_name, {inputValue: cluster_name},
+      --main_jar_file_uri, {inputValue: main_jar_file_uri},
+      --main_class, {inputValue: main_class},
+      --args, {inputValue: args},
+      --hadoop_job, {inputValue: hadoop_job},
+      --job, {inputValue: job},
+      --wait_interval, {inputValue: wait_interval}
+    ]
+    env:
+      KFP_POD_NAME: "{{pod.name}}"
+    fileOutputs:
+      job_id: /tmp/kfp/output/dataproc/job_id.txt
```

Comment:
same as above, does it make sense to output the job json as the output?

Reply:
ditto here.